app/src/main/java/com/duckduckgo/widget/SearchWidget.kt

```diff
@@ -23,12 +23,22 @@ import android.content.Context
 import android.widget.RemoteViews
 import com.duckduckgo.app.browser.BrowserActivity
 import com.duckduckgo.app.browser.R
+import com.duckduckgo.app.global.DuckDuckGoApplication
+import com.duckduckgo.app.statistics.pixels.Pixel.PixelName.*
+import com.duckduckgo.app.widget.ui.supportsAutomaticWidgetAdd
 
 
 class SearchWidgetLight : SearchWidget(R.layout.search_widget_light)
 
 open class SearchWidget(val layoutId: Int = R.layout.search_widget) : AppWidgetProvider() {
 
+    override fun onEnabled(context: Context) {
+        val application = context.applicationContext as? DuckDuckGoApplication
```

Comment:
It would be much cleaner to use DI here. Can you think of a way to get DI working for the widgets?

Reply:
This _might_ work?

    override fun onReceive(context: Context, intent: Intent?) {
        inject(context)
        super.onReceive(context, intent)
    }

    private fun inject(context: Context) {
        (context.applicationContext as DuckDuckGoApplication).daggerAppComponent.inject(this)
    }


Finally, you'd need to add this line to the AppComponent class

    fun inject(searchWidget: SearchWidget)

so that it would become 

    interface AppComponent : AndroidInjector<DuckDuckGoApplication> {
    fun inject(searchWidget: SearchWidget)

    @Component.Builder
    interface Builder {

        @BindsInstance
        fun application(application: Application): AppComponent.Builder

        fun build(): AppComponent
    }
}

Reply:
Not sure how desired that is, but would probably work! 😓

Reply:
Good to know! I'll skip it for now as it's not exactly reducing work _yet_ but it's useful to know for next time.